common/src/main/java/io/netty/util/concurrent/DefaultPromise.java

```diff
@@ -86,7 +86,7 @@ public DefaultPromise(EventExecutor executor) {
     /**
      * See {@link #executor()} for expectations of the executor.
      */
-    protected DefaultPromise() {
+    public DefaultPromise() {
```

Comment:
Honestly I think we need to fix the in another way...  I am not comfortable with changing the access levels and so make it "public api" (this comment is true for all the changes)